### DISCUSSION TITLE
Is there any configuration to send Sender Report of RTCP protocol?

### DISCUSSION THREAD

**Discussion by sautvaste** [bluenviron/mediamtx#1273] [Questions] [answered]
I publish a rtsp stream, but there is no sender report has been sent, I need to get NTP timestamp from the packet on the client side, it there any resolve?

Thanks in advance!

**Comment by aler9** [comment, marked as answer]
Hello, NTP timestamps are something i've been working on a lot recently. Actually, NTP timestamps of the original source are lost, since sender reports are re-generated by the server as you mentioned, but nonetheless packets are marked with an absolute timestamp that is correctly propagated to readers through RTCP sender reports and even through RTP-Info headers (it's even propagated when a stream is read with HLS, thanks to the `EXT-X-PROGRAM-DATE-TIME` attribute).

Therefore, what is missing is the ability to

1. get NTP timestamp of original packets
2. use that timestamp inside the server, replacing it only in case it's missing, and also test what happens when there's a big discrepancy between a reader's NTP and a camera's NTP

Point 1 is being tracked here:

https://github.com/aler9/gortsplib/issues/159

And point 2 is being tracked here:

#1300

Use these two issues to provide further contributions.
